src/Servers/Kestrel/Transport.Quic/src/QuicConnectionFactory.cs

```diff
@@ -40,13 +40,8 @@ public async ValueTask<MultiplexedConnectionContext> ConnectAsync(EndPoint endPo
             {
                 throw new NotSupportedException($"{endPoint} is not supported");
             }
-            if (_transportContext.Options.Alpn == null)
-            {
-                throw new InvalidOperationException("QuicTransportOptions.Alpn must be configured with a value.");
```

Comment:
Does Alpn get checked in the runtime now?

Reply:
Yes, MsQuic enforces this:
```
System.Exception: At least one SslApplicationProtocol value must be present in SslClientAuthenticationOptions or SslServerAuthenticationOptions.
   at System.Net.Quic.Implementations.MsQuic.Internal.SafeMsQuicConfigurationHandle.Create(QuicOptions options, QUIC_CREDENTIAL_FLAGS flags, X509Certificate certificate, SslStreamCertificateContext certificateContext, List`1 alpnProtocols) in System.Net.Quic.dll:token 0x60001e3+0x17
```

I've also changed our HTTP/3 implementation to always set ALPN via the Ssl options so users don't need to worry about it anymore.